init.cmd

```diff
@@ -0,0 +1,56 @@
+@ECHO OFF
+
+REM Don't run inside of devenv shells
+IF NOT "%VSCMD_VER%"=="" (
+  ECHO ERROR: Do not run from a Developer Command prompt.
+  ECHO Use a plain shell and try again.
+  EXIT /b 10
+)
+
+REM Don't run twice
+IF NOT "%VFS_DEVSHELL%"=="" (

```

Comment:
You would only need to rerun if envvars changed, or to pick up a new macro. Let's try it like this, since it means we don't need to worry with environment upgrades. If it proves clunky we'll revisit.

Reply:
OK but when that happens I'll probably just comment out this line rather than restart my shell :-)